### ISSUE TITLE
Setting TextureRect.texture does not update the node's size in the same frame.

### ISSUE THREAD

**Issue by vicguedez** [godotengine/godot#89258]
### Tested versions

Godot 4.3.dev4.official

### System information

Pop!_OS 22.04 LTS - 64 Bits

### Issue description

Doing `TextureRect.texture = load("path.image")` does not update the node's size until the next frame. Which makes things like replacing the texture in the same call behave unexpectedly if you are assuming that doing `TextureRect.texture = null` resets the `size` to zero.

Workaround is to wait for the next frame for it to properly reset the `size`.

### Steps to reproduce

Reproducible in the MRP provided.

### Minimal reproduction project (MRP)

[project.zip](https://github.com/godotengine/godot/files/14528477/project.zip)

**Comment by AThousandShips** [comment]
This is the case generally for controls I believe, for performance reasons, might be documented better though

**Comment by vicguedez** [comment]
> This is the case generally for controls I believe, for performance reasons, might be documented better though

Yes I understand and adding a warning (maybe a note on the texture prop directly) is definitely needed.

**Comment by timothyqiu** [comment]
If it's only for performance reasons, I think maybe it's okay to provide a `force_update()` method to trigger the update manually in the current frame.

**Comment by hippogyz** [comment]
In my case, I need manually set `size` of TextureRect after `texture` override. 

But setting `size` does not work no matter immediately or 1-frame-later or 2-frame later after overriding `texture`. 

Fortunately, I found setting `size` twice can solve this, just like
```
 texture = my_texture;
 size = my_size;
 size = my_size;  
```

I don't know why, but it just works. XD
